Markdown options (#330)
CloudCannon/platform-documentation: articles/configure-your-markdown-engine.mdx

related_articles: []
related_links:
---

> CloudCannon needs to parse Markdown content into HTML in order to preview and edit it. It's important for CloudCannon to save Markdown in the same format expected by your SSG, so that your content is built correctly.

Reviewer: ```suggestion
CloudCannon needs to parse Markdown content into HTML to preview and edit it. Your Markdown must be in a format recognized by your SSG so that CloudCannon can build your content.
```

You can use the top-level `markdown` key in your [global configuration file](/documentation/articles/create-a-global-configuration-file/) to define how CloudCannon should parse and save your Markdown content.

<comp.MultiCodeBlock language="YAML" translate_into={["json", "JavaScript"]} source="cloudcannon.config.yml">
```
markdown:
  engine: 'commonmark'
  options:
    table: true
    code_block_fences: '```'
```
</comp.MultiCodeBlock>

> CloudCannon supports the CommonMark and Kramdown specifications. The implementations for these are `markdown-it` (JavaScript) and `kramdown` (Ruby). You can use any Markdown engine you want when building your site. This article only concerns the way CloudCannon parses and saves your Markdown while you're editing. 

Reviewer: ```suggestion
CloudCannon supports CommonMark and Kramdown using `markdown-it` (JavaScript) and `kramdown` (Ruby). You can use any Markdown engine you want when building your site. This article only concerns the way CloudCannon parses and saves your Markdown while you're editing. 
```

 <comp.DataReferenceRow label="options.subscript" type_markdown="boolean">
  If `true`, subscript elements (`<sub>`) are wrapped with tildes (e.g. \~example\~), instead of being output as HTML.
  
  Enable this option only if your SSG's Markdown engine recognizes this syntax.
 </comp.DataReferenceRow>
 <comp.DataReferenceRow label="options.superscript" type_markdown="boolean">
  If `true`, subscript elements (`<sup>`) are wrapped with carets (e.g. ^example^), instead of being output as HTML.
  
  Enable this option only if your SSG's Markdown engine recognizes this syntax.
 </comp.DataReferenceRow>

 <comp.DataReferenceRow label="options.attributes" type_markdown="boolean">
  If `true`, HTML attributes will be saved in Markdown format. Otherwise, elements with attributes may be converted to HTML to preserve the attributes.
  
  Enable this option only if your SSG's Markdown engine recognizes this syntax.
 </comp.DataReferenceRow>
 <comp.DataReferenceRow label="options.gfm" type_markdown="boolean">
  Enables extra parsing options consistent with GitHub Flavored Markdown. 
  
  Defaults to `true`.
 </comp.DataReferenceRow>
</comp.DataReference>

### Display-only options

> Some Markdown options perform transformations of your content which make sense when rendering for display, but which you wouldn't want to save back to your content files. For this reason, some options within `markdown.options` are used for display, but are deliberately ignored when saving content back to your files.

Reviewer: ```suggestion
Some Markdown options transform your content to render a display but should not affect the saved content in your files. For this reason, CloudCannon will use some options in `markdown.options` are used for display only.
```

> These options are used for [visual data previews](/documentation/articles/what-are-visual-data-previews/) in the [Visual Editor](/documentation/articles/the-visual-editor/), [comments on inputs](/documentation/articles/configure-your-inputs/#comment), and [descriptions on collections](/documentation/articles/define-your-collections/#description).

Reviewer: ```suggestion
Display-only options facilitate [visual data previews](/documentation/articles/what-are-visual-data-previews/) in the [Visual Editor](/documentation/articles/the-visual-editor/), [comments on inputs](/documentation/articles/configure-your-inputs/#comment), and [descriptions for collections](/documentation/articles/define-your-collections/#description).
```

<comp.Notice info_type="important">
  If your site uses [visual data previews](/documentation/articles/what-are-visual-data-previews/) to enable live editing from your rich text frontmatter fields, CloudCannon will use all the above options when parsing your Markdown and passing it into the Visual Editor.

>   CloudCannon does not automatically communicate your Markdown settings with any libraries you might use to render visual data previews within the editor. If those libraries need to reparse your Markdown and do not also read your Markdown settings, some content may be rendered inconsistently in the previews.

Reviewer: ```suggestion
  CloudCannon does not automatically communicate your Markdown settings with any libraries you might use to render visual data previews within the editor. If those libraries need to reparse your Markdown and do not also read your Markdown settings, CloudCannon may inconsistently render some content in the previews.
```

### Display-only options

> Some Markdown options transform your content to render displays but should not affect the saved content in your files. For this reason, CloudCannon will use some options in `markdown.options` for display only.

Reviewer: ```suggestion
Display-only Markdown options in `markdown.options` can transform your content to render displays but cannot affect the saved content in your files.
```
